src/Components/OATHeader/internal/ModalSaveCurrentProjectAndClear.tsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import {
+    Text,
+    ActionButton,
+    FontIcon,
+    PrimaryButton,
+    Stack
+} from '@fluentui/react';
+import { useTranslation } from 'react-i18next';
+import {
+    OATDataStorageKey,
+    OATFilesStorageKey
+} from '../../../Models/Constants';
+import { getHeaderStyles, getPromptTextStyles } from '../OATHeader.styles';
+import { IOATEditorState } from '../../../Pages/OATEditorPage/OATEditorPage.types';
+
+interface IModal {
+    resetProject?: () => void;
+    setModalBody?: React.Dispatch<React.SetStateAction<string>>;
+    setModalOpen?: React.Dispatch<React.SetStateAction<boolean>>;
+    state?: IOATEditorState;
+}
+
+export const ModalSaveCurrentProjectAndClear = ({
+    resetProject,
+    setModalOpen,
+    setModalBody,
+    state
+}: IModal) => {
+    const { t } = useTranslation();
+    const headerStyles = getHeaderStyles();
+    const promptTextStyles = getPromptTextStyles();
+    const { projectName } = state;
+
+    const handleOnSave = () => {
+        const files = JSON.parse(localStorage.getItem(OATFilesStorageKey));
```

Comment:
We parse from storage *a lot*. Can't we keep this in state? And manage it there?